packages/mock/src/fulcio/ctlog.ts

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2023 The Sigstore Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { Crypto, CryptoKey } from '@peculiar/webcrypto';
+import * as asn1js from 'asn1js';
+import * as bs from 'bytestreamjs';
+import * as pkijs from 'pkijs';
+import * as pvutils from 'pvutils';
+import {
+  DIGEST_SHA256,
+  KEY_ALGORITHM_ECDSA_P256,
+  SIGNING_ALGORITHM_ECDSA_SHA256,
+} from '../constants';
+
+export interface CTLog {
+  publicKey: Buffer;
+  logID: Buffer;
+  log: (
+    certificate: ArrayBuffer,
+    issuerID: ArrayBuffer
+  ) => Promise<ArrayBuffer>;
+}
+
+export async function initializeCTLog(): Promise<CTLog> {
+  const crypto = new Crypto();
+
+  const { publicKey, privateKey } = await crypto.subtle.generateKey(
+    KEY_ALGORITHM_ECDSA_P256,
+    true,
+    ['sign', 'verify']
+  );
+
+  const publicKeyBytes = await crypto.subtle
+    .exportKey('spki', publicKey)
+    .then((bytes) => Buffer.from(bytes));
+
+  const logID = await crypto.subtle
+    .digest({ name: DIGEST_SHA256 }, publicKeyBytes)
+    .then((bytes) => Buffer.from(bytes));
+
+  return new CTLogImpl({
+    publicKey: publicKeyBytes,
+    privateKey,
+    logID,
+    crypto,
+  });
+}
+
+interface CTLogOptions {
+  publicKey: Buffer;
+  logID: Buffer;
+  privateKey: CryptoKey;
+  crypto: Crypto;
+}
+
+class CTLogImpl implements CTLog {
+  public readonly publicKey: Buffer;
+  public readonly logID: Buffer;
+  private privateKey: CryptoKey;
+  private crypto: Crypto;
+
+  constructor(options: CTLogOptions) {
+    this.publicKey = options.publicKey;
+    this.privateKey = options.privateKey;
+    this.logID = options.logID;
+    this.crypto = options.crypto;
+  }
+
+  public async log(
+    certificate: ArrayBuffer,
+    issuerID: ArrayBuffer
+  ): Promise<ArrayBuffer> {
+    const version = 0;
+    const timestamp = new Date();
+    const logID = this.logID;
+
+    // A bit of a hack to get the TBS bytes
+    const tbs = pkijs.Certificate.fromBER(certificate).encodeTBS().toBER();
+
+    // Calculate the SCT signature
```

Comment:
Could we link to something where this is implemented in fulcio/ctlog for reference? Thinking down the line there might be some drift between implementations so having a way to find where things are implemented could be helpful.